src/puppetlabs/pcp/client.clj

```diff
@@ -246,33 +192,24 @@
   [client :- Client timeout :- s/Num]
   (let [{:keys [websocket-connection]} client]
     (try
-      (if (deref @websocket-connection timeout nil)
-        client
-        nil)
+      (when (deref @websocket-connection timeout nil)
+        client)
       (catch java.util.concurrent.ExecutionException exception
-        (log/debug exception (i18n/trs "exception while waiting for a connection; not connected"))
-        nil))))
-
-(s/defn -wait-for-association :- (s/maybe Client)
-  "Waits until a client is associated. If timeout is hit, or the association doesn't work, returns falsey"
-  [client :- Client timeout :- s/Num]
-  (let [{:keys [associate-response]} client]
-    (if (deref @associate-response timeout nil)
-      client
-      nil)))
+        (log/debug exception (i18n/trs "exception while waiting for a connection; not connected"))))))
 
 (s/defn ^:private -send! :- s/Bool
   "Send a message across the websocket session"
   [client :- Client message :- message/Message]
   (let [{:keys [identity websocket-connection]} client]
-    (if-not (-associated? client)
-      (throw+ {:type ::not-associated})
+    (if-not (connected? client)
+      (do (log/debug (i18n/trs "refusing to send message on unconnected session"))
+          (throw+ {:type ::not-connected}))
       (try
         (ws/send-msg @@websocket-connection
                      (message/encode (assoc message :sender identity)))
         (catch java.util.concurrent.ExecutionException exception
           (log/debug exception (i18n/trs "exception on the connection while attempting to send a message"))
-          (throw+ {:type :not-connected}))))
+          (throw+ {:type ::not-connected}))))
```

Comment:
the namespaced keywords probably come from this doc https://tickets.puppetlabs.com/browse/CT-18, which afaik is where the practice originated.